lib/reek/tree_walker.rb

```diff
@@ -53,10 +54,6 @@ def walk
     # RootContext -> children: 1 ModuleContext -> children: 1 MethodContext
     #
     # @return [Reek::Context::RootContext] tree of nested contexts
-    def context_tree
-      @context_tree ||= process(exp)
```

Comment:
I said it before and I'll say it again: We are heavily overusing this pattern in Reek and so it was here.